### ISSUE TITLE
Overriding property makes it appear twice in schema (Crash)

### ISSUE THREAD

**Issue by amkomatz** [realm/realm-cocoa#6303]
## Important
I did not notice this to be an issue in earlier versions of Realm, only after updating to v3.19.0 and Xcode 11. I have attached a project, it should crash on launch.

Important note on the attached project: before running, the project, do `pod install` in the terminal. To upload the project, I had to deintegrate the pods.

[RealmCrash.zip](https://github.com/realm/realm-cocoa/files/3725887/RealmCrash.zip)

## Goals
Properties on a Realm object should be able to be overridden in subclasses in order to add new functionality, property observers, etc.

## Expected Results
Overridden properties should only be in the object's schema once, and should not result in a crash.

## Actual Results
Overriding a property results in that property appearing in the object's schema twice. This occurs when overriding with get/set, or willSet/didSet.

"Terminating app due to uncaught exception 'RLMException', reason: 'Property 'id' is declared multiple times in the class hierarchy of 'UserModel'"

## Steps to Reproduce
Simply create one model with a property, and create a subclass that adds `willSet` to that property. 

If you also turn on exception breakpoints in Xcode, and run the app, it should crash in `RealmObjectShcema.mm` on line `148`. If you run the command `po schema`, you will see that the property is being recorded twice:

```
(lldb) po schema
UserModel {
	id {
		type = int;
		objectClassName = (null);
		linkOriginPropertyName = (null);
		indexed = NO;
		isPrimary = NO;
		array = NO;
		optional = NO;
	}
	id {
		type = int;
		objectClassName = (null);
		linkOriginPropertyName = (null);
		indexed = NO;
		isPrimary = NO;
		array = NO;
		optional = NO;
	}
}
```

## Code Sample
```swift
public class ObjectModel: Object {
    
    public override class func shouldIncludeInDefaultSchema() -> Bool { false }
    
    @objc dynamic public var id: Int = 0
}

public class UserModel: ObjectModel {
    
    public override class func shouldIncludeInDefaultSchema() -> Bool { true }
    
    public override var id: Int {
        willSet {
            print(newValue)
        }
    }
}
```

## Version of Realm and Tooling
Realm framework version: 3.19.0
Xcode version: 11.1
iOS/OSX version: 13.0

**Comment by tgoyne** [comment]
`willSet` and `didSet` don't work for managed Realm objects to begin with, so it seems unlikely that trying to override them in a subclass would do anything useful.

**Comment by amkomatz** [comment]
@tgoyne I can confirm that `willSet` and `didSet` work perfectly fine when setting the property from within Swift, which is the functionality I need (I am not working with Realm sync in my project).

Also, that was just an example. The same issue is happening if you override `get`/`set` as well. The issue here is `override` :)

**Comment by amkomatz** [comment]
I can also confirm that this is happening even if `shouldIncludeInDefaultSchema()` returns `false`.

**Comment by leemaguire** [comment]
Closing this issue as it has been inactive for a while. Please update to the latest version of Realm and if you have any other questions please open a new issue. Thanks.
